cellrank/kernels/_transport_map_kernel.py

```diff
@@ -229,80 +226,227 @@ def compute_transition_matrix(
 
         return self
 
+    @classmethod
+    def from_moscot(
+        cls,
+        problem: "CompoundProblem",
+        allow_subset: bool = False,
+        sparsify: bool = False,
+        sparsify_kwargs: Mapping[str, Any] = types.MappingProxyType({}),
+        **kwargs: Any,
+    ) -> "TransportMapKernel":
+        """Construct the kernel from :mod:`moscot` :cite:`klein:23`.
+
+        Parameters
+        ----------
+        problem
+            A :mod:`moscot` problem.
+        allow_subset
+            Whether to subset :attr:`~moscot.base.problems.CompoundProblem.adata` if the solutions
+            do not completely partition it.
+        sparsify
+            Whether to sparsify the transport maps using :meth:`~moscot.base.output.BaseSolverOutput.sparsify`.
+        sparsify_kwargs
+            Keyword arguments for the sparsification.
+        kwargs
+            Additional keyword arguments for :class:`~cellrank.kernels.TransportMapKernel`.
+
+        Returns
+        -------
+        The kernel.
+
+        Notes
+        -----
+        If ``allow_subset = True``, :attr:`adata` will be a view.
+        """
+        from moscot.problems.time import TemporalMixin
+        from moscot.problems.space import SpatialAlignmentMixin
+        from moscot.utils.subset_policy import (
+            StarPolicy,
+            SequentialPolicy,
+            ExternalStarPolicy,
+        )
+
+        if not problem.solutions:
+            raise RuntimeError(
+                "Problem contains no solutions. Please run `problem.solve(...)` first."
+            )
+
+        if isinstance(problem, TemporalMixin):
+            batch_key = problem.temporal_key
+        elif isinstance(problem, SpatialAlignmentMixin):
```

Comment:
Is this neccesary? The `batch_key` in the `SpatioTemporalProblem` should be irrelevant, and be the same as the `time_key`.

Reply:
`batch_key` in the `SpatioTemporalKernel` will be the `time_key`, since it also subclasses the `TemporalMixin`, for which I check first.
Maybe the `batch_key` in the context of the kernel is not the best name, but couldn't find anything better. Maybe `covariate_key`? Pinging @Marius1311 for opinion.

Reply:
Yeah doesn't make sense I think.
I mean it's a detail, but all moscot problems which make sense to use in CellRank, inherit from the `TemporalMixin` and hence have a `temporal_key`.

Reply:
I agree, it does make make sense to allow this functionality for the`SpatialAligment` problem. The `TransportMap` Kernel should be coupled to a notion of time. CellRank is all about cellular dynamic, so we need a temporal element.